app/src/components/map.jsx

```diff
@@ -26,144 +25,187 @@ class Map extends Component {
     this.state = {
       overlay: null,
       addedLayers: [],
-      ite: tmlnMinDate,
+      currentTimestamp: TIMELINE_MIN_DATE,
       lastCenter: null,
-      filters: {
-        startDate: tmlnMinDate,
-        endDate: tmlnMaxDate,
-        flag: ''
-      }
+      playbackLength: 1,
+      vesselLayerTransparency: 1,
+      currentVesselInfo: {},
+
+      leftHandlerPosition: 0,
+      rightHandlerPosition: 0,
+      timeBarWidth: 0,
     };
   }
 
+  /**
+   * Zoom change handler
+   * Enforces min and max zoom levels
+   * Resets vessel layer data on change
+   */
   onZoomChanged() {
-    const ZOOM = this.refs.map.props.map.getZoom();
-    if (ZOOM < maxZoomLevel) {
-      this.refs.map.props.map.setZoom(maxZoomLevel);
+    const zoom = this.refs.map.props.map.getZoom();
+    if (zoom < MIN_ZOOM_LEVEL) {
+      this.refs.map.props.map.setZoom(MIN_ZOOM_LEVEL);
+    }
+    if (zoom > MAX_ZOOM_LEVEL) {
+      this.refs.map.props.map.setZoom(MAX_ZOOM_LEVEL);
     }
-    this.setState({zoom: ZOOM});
-    this.state.overlay.resetData();
+    this.setState({zoom: zoom});
+    this.state.overlay.resetPlaybackData();
   }
 
-  moveTimeline(e) {
+  /**
+   * TODO: review
+   *
+   * @param event
+   */
+  moveTimeline(event) {
     this.timelinerange = document.getElementById('timeline_handler');
-    this.timelineStart(this.timelinerange.style.width = (e.clientX - 60) + 'px');
+    this.playbackStart(this.timelinerange.style.width = (event.clientX - 60) + 'px');
   }
 
-  handlerMoved(tick, ev) {
+  handlerMoved(tick, event) {
     let target = null;
-    const TIMELINESCOPE = document.getElementById('timeline_handler').parentElement;
-    const ABSMAXMOMENT = new Date(tmlnMaxDate).getTime() - new Date(tmlnMinDate).getTime();
-    let PERCENTAGE = null;
+    let startDate = this.props.filters.startDate;
+    let endDate = this.props.filters.endDate;
+
+    const timelineScope = document.getElementById('timeline_handler').parentElement;
+    const absMaxMoment = new Date(endDate).getTime() - new Date(startDate).getTime();
     if (tick === 1) {
       target = document.getElementById('dateHandlerLeft');
     } else {
       target = document.getElementById('dateHandlerRight');
     }
-    PERCENTAGE = ((target.getBoundingClientRect().left - TIMELINESCOPE.getBoundingClientRect().left) * 100) / TIMELINESCOPE.offsetWidth;
+    let percentage = ((target.getBoundingClientRect().left - timelineScope.getBoundingClientRect().left) * 100) / timelineScope.offsetWidth;
```

Comment:
Why `percentage` is not a const anymore and why the consts are not uppercased as before?